pyquil/paulis.py

```diff
@@ -250,6 +270,12 @@ def __init__(self, terms):
     def __str__(self):
         return " + ".join([str(term) for term in self.terms])
 
+    def __getitem__(self, item):
+        return self.terms[item]
```

Comment:
Do you want to catch `TypeError`s and give raise a more specific error? Currently `(sZ(1)-sX(1))['foo']` gives
```
TypeError: list indices must be integers, not str
```